datalad/core/local/diff.py

```diff
@@ -329,7 +328,9 @@ def _diff_ds(ds, fr, to, constant_refs, recursion_level, origpaths, untracked,
             init=diff_state,
             eval_availability=annexinfo in ('availability', 'all'),
             ref=to)
-        if fr != to:
+        # if `fr` is None, we compare against a preinit state, and
+        # nothing needs to be done
```

Comment:
Might be worth being a bit more explicit why "nothing needs to be done" (we just got all we need to know, since current state of affairs *is* the diff to preinit). Took me a while to realize - `diff` is just not the easiest thing to have a quick look at.
That is if I'm not mistaken, of course. If I am wrong then it's even more important ;-)

Reply:
I am not sure I understand. "preinit" is before the repository existed. That state is "nothing" and it is identical across all datasets. So in order to figure everything out about this state, nothing needs to be done. I don't know how to change that comment to address your question. Feel free to push a better comment.